Makefile

```diff
@@ -0,0 +1,47 @@
+ifeq ($(OS),Windows_NT)
```

Comment:
rather than using Make, could we stick to CMake and implement the same by based on CMake?

Reply:
This is the approach that https://github.com/neovim/neovim uses.
Running `make` is much easier to remember to setup the project.
Running from the command line `cmake` a couple of long commands to setup the build system.
This is useful in CI scripts and docker to simplify the setup.

Reply:
@chinwobble, I see that Neovim does the same thing. The CMakePresets.json should be sufficient to handle the task: you open the folder with an IDE (or build from the command line), set the preset name, and the build or test will run with the correct command line, which may be complex, but the end user does not need to know that.

To keep this template simple, I think Make should not be used. The main goal is to use CMake, which is truly multi platform. While Make might work on Windows, in my experience, it only functions well with Cygwin or a Unix-like build environment. Therefore, a template that suggests using CMake but then relies on Make as the primary driver doesn't seem ideal, does it?

Reply:
@chinwobble, this seems to be my only concern. If we could leverage the CMakePresets.json feature and avoid using Make, that would be ideal. However, it might not be possible. Could you provide an example of something the Makefile does that cannot be achieved using CMakePresets?

Reply:
The two commands that are added to the Makefile are 
```bash
make format
make lint (clang-tidy)
```

Maybe this can be accomplished in the `CMakeLists.txt` file but I'm not sure what the `best` way to do that is.
I'm new to CMake and have only been learning C++ since 2 weeks ago.
Since `clang-tidy` takes awhile to run, we don't want to run that every time, we want a separate command for that (maybe in CMake terms that is a custom target?)
Perhaps `format` can be run as a CMake workflow but I haven't seen any examples of that. I've looked in many repos from
https://github.com/google/highway
https://github.com/microsoft/cpp-async

No standard way to do this.